### ISSUE TITLE
Help with Finetune

### ISSUE THREAD

**Issue by EmailScraper** [OpenBMB/MiniCPM-V#242]
Hi, team.
I fine tuned with 15 <image, Q&A> pairs of one document type. (12 pairs as train, 3 as test)
After 500 epoches, both of training and eval_loss are nearly zero like following.

> 
    {'loss': 0.0, 'grad_norm': 7.342097887885757e-06, 'learning_rate': 1e-06, 'epoch': 500.0}
    {'eval_loss': 7.8905128475526e-07, 'eval_runtime': 2.179, 'eval_samples_per_second': 1.377, 'eval_steps_per_second': 0.459, 'epoch': 500.0}  

But even I pick up image and ask same questions from training set, still get different result from training set like following
- Training Data
>
    {
          "role": "user",
          "content": "What is the address breakdown city?"
        },
        {
          "role": "assistant",
          "content": "CALWELL"
        },
        {
          "role": "user",
          "content": "What is the address breakdown street?"
        },
        {
          "role": "assistant",
          "content": "65 MARTLEY CCT"
        },
        {
          "role": "user",
          "content": "What is the address?"
        },
        {
          "role": "assistant",
          "content": "65 MARTLEY CCT, CALWELL, ACT, 2905"
        },

- Model Output
>
    <User>: What is the address breakdown city?
    <Assistant>: 65 MARTLEY CCT
    <User>: What is the address breakdown street?
    <Assistant>: 65 MARTLEY CCT, suburb CALWELL ACT 2905.
    <User>: What is the address?
    <Assistant>: 65 MARTLEY CCT CALWELL ACT 2905

I used batch_size 8, and model_max_length 1024.

I loaded model like following.
>
    model_path = '/home/paperspace/.../OpenBmb/MiniCPM-V/finetune/output/output_minicpmv2_lora/checkpoint-1000'
    model = AutoPeftModelForCausalLM.from_pretrained(model_path, device_map="auto", trust_remote_code=True).to(dtype=torch.float16)
    model = model.to(device=device)
    tokenizer = AutoTokenizer.from_pretrained(model_path, trust_remote_code=True)
    model.eval()

Tensorboard
![image](https://github.com/OpenBMB/MiniCPM-V/assets/166853562/c10c9f08-f34f-463c-9c5b-281f5feb398a)

**Any help would be appreciated to get correct finetuning result.**

**Comment by udrs** [comment]
插眼

**Comment by WAILMAGHRANE** [comment]
> Hi, team. I fine tuned with 15 <image, Q&A> pairs of one document type. (12 pairs as train, 3 as test) After 500 epoches, both of training and eval_loss are nearly zero like following.
> 
> > 
> 
> ```
> {'loss': 0.0, 'grad_norm': 7.342097887885757e-06, 'learning_rate': 1e-06, 'epoch': 500.0}
> {'eval_loss': 7.8905128475526e-07, 'eval_runtime': 2.179, 'eval_samples_per_second': 1.377, 'eval_steps_per_second': 0.459, 'epoch': 500.0}  
> ```
> 
> But even I pick up image and ask same questions from training set, still get different result from training set like following
> 
> * Training Data
> 
> > 
> 
> ```
> {
>       "role": "user",
>       "content": "What is the address breakdown city?"
>     },
>     {
>       "role": "assistant",
>       "content": "CALWELL"
>     },
>     {
>       "role": "user",
>       "content": "What is the address breakdown street?"
>     },
>     {
>       "role": "assistant",
>       "content": "65 MARTLEY CCT"
>     },
>     {
>       "role": "user",
>       "content": "What is the address?"
>     },
>     {
>       "role": "assistant",
>       "content": "65 MARTLEY CCT, CALWELL, ACT, 2905"
>     },
> ```
> 
> * Model Output
> 
> > 
> 
> ```
> <User>: What is the address breakdown city?
> <Assistant>: 65 MARTLEY CCT
> <User>: What is the address breakdown street?
> <Assistant>: 65 MARTLEY CCT, suburb CALWELL ACT 2905.
> <User>: What is the address?
> <Assistant>: 65 MARTLEY CCT CALWELL ACT 2905
> ```
> 
> I used batch_size 8, and model_max_length 1024.
> 
> I loaded model like following.
> 
> > 
> 
> ```
> model_path = '/home/paperspace/.../OpenBmb/MiniCPM-V/finetune/output/output_minicpmv2_lora/checkpoint-1000'
> model = AutoPeftModelForCausalLM.from_pretrained(model_path, device_map="auto", trust_remote_code=True).to(dtype=torch.float16)
> model = model.to(device=device)
> tokenizer = AutoTokenizer.from_pretrained(model_path, trust_remote_code=True)
> model.eval()
> ```
> 
> Tensorboard ![image](https://private-user-images.githubusercontent.com/166853562/337055397-c10c9f08-f34f-463c-9c5b-281f5feb398a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTc3ODk1NzcsIm5iZiI6MTcxNzc4OTI3NywicGF0aCI6Ii8xNjY4NTM1NjIvMzM3MDU1Mzk3LWMxMGM5ZjA4LWYzNGYtNDYzYy05YzViLTI4MWY1ZmViMzk4YS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNjA3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDYwN1QxOTQxMTdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hZDI0OGNhZWY5MDE1NDhjZTg1NzgxYzA3YzE0NGRkODY3ZTU2MTMwNjI2YzQ0MjNkMmU2MjFiMjBmNDgwY2MzJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.w9wwHFHKvrdENH0Fe7RBhbVH8IvqRlIAnalQSHLmNls)
> 
> **Any help would be appreciated to get correct finetuning result.**

Can you give me the code ?

**Comment by JinQiangWang2021** [comment]
> Hi, team. I fine tuned with 15 <image, Q&A> pairs of one document type. (12 pairs as train, 3 as test) After 500 epoches, both of training and eval_loss are nearly zero like following.
> 
> > 
> 
> ```
> {'loss': 0.0, 'grad_norm': 7.342097887885757e-06, 'learning_rate': 1e-06, 'epoch': 500.0}
> {'eval_loss': 7.8905128475526e-07, 'eval_runtime': 2.179, 'eval_samples_per_second': 1.377, 'eval_steps_per_second': 0.459, 'epoch': 500.0}  
> ```
> 
> But even I pick up image and ask same questions from training set, still get different result from training set like following
> 
> * Training Data
> 
> > 
> 
> ```
> {
>       "role": "user",
>       "content": "What is the address breakdown city?"
>     },
>     {
>       "role": "assistant",
>       "content": "CALWELL"
>     },
>     {
>       "role": "user",
>       "content": "What is the address breakdown street?"
>     },
>     {
>       "role": "assistant",
>       "content": "65 MARTLEY CCT"
>     },
>     {
>       "role": "user",
>       "content": "What is the address?"
>     },
>     {
>       "role": "assistant",
>       "content": "65 MARTLEY CCT, CALWELL, ACT, 2905"
>     },
> ```
> 
> * Model Output
> 
> > 
> 
> ```
> <User>: What is the address breakdown city?
> <Assistant>: 65 MARTLEY CCT
> <User>: What is the address breakdown street?
> <Assistant>: 65 MARTLEY CCT, suburb CALWELL ACT 2905.
> <User>: What is the address?
> <Assistant>: 65 MARTLEY CCT CALWELL ACT 2905
> ```
> 
> I used batch_size 8, and model_max_length 1024.
> 
> I loaded model like following.
> 
> > 
> 
> ```
> model_path = '/home/paperspace/.../OpenBmb/MiniCPM-V/finetune/output/output_minicpmv2_lora/checkpoint-1000'
> model = AutoPeftModelForCausalLM.from_pretrained(model_path, device_map="auto", trust_remote_code=True).to(dtype=torch.float16)
> model = model.to(device=device)
> tokenizer = AutoTokenizer.from_pretrained(model_path, trust_remote_code=True)
> model.eval()
> ```
> 
> Tensorboard ![image](https://private-user-images.githubusercontent.com/166853562/337055397-c10c9f08-f34f-463c-9c5b-281f5feb398a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTc5MDg0NzksIm5iZiI6MTcxNzkwODE3OSwicGF0aCI6Ii8xNjY4NTM1NjIvMzM3MDU1Mzk3LWMxMGM5ZjA4LWYzNGYtNDYzYy05YzViLTI4MWY1ZmViMzk4YS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNjA5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDYwOVQwNDQyNTlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1jODI5MjgxODI2OTViMjE5Y2U4ZTU2NjJkNGJlMzcyMGFhNTdiM2FjYjQxM2MxMTMwMzhmNzQ1NWVjOGYxYWFiJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.SjNwzOO_6hGQH6bAYrBlutoOPRgPaHJbTLgWG1QYhKU)
> 
> **Any help would be appreciated to get correct finetuning result.**

@EmailScraper  Hello, may I ask if you have trained with images? When I fine tuned, it kept showing loading the data. Have you encountered this? May I inquire with you? Thank you
<img width="902" alt="image" src="https://github.com/OpenBMB/MiniCPM-V/assets/9883881/3571296a-da36-4eb2-8b5e-4580ec29090f">

**Comment by LDLINGLINGLING** [comment]
我认为有可能是你的训练数据集太小了，而且训练epoch设置为500，又太多了导致模型严重过拟合

**Comment by EmailScraper** [comment]
Hi, @LDLINGLINGLING , thanks for your reply.
Btw, I  just pick up image and ask same questions from training set, still get different result from training set.
Is it possible?
